local-container-endpoints/handlers/credentials_handler.go

```diff
@@ -195,12 +196,22 @@ func (service *CredentialService) getTemporaryCredentials() (*CredentialResponse
 			SecretAccessKey: credVal.SecretAccessKey,
 			Token:           credVal.SessionToken,
 		}
+
 		expiration, err := service.currentSession.Config.Credentials.ExpiresAt()
-		// It is valid for a credential provider to not return an expiration
-		// TODO: Check if expiration is optional from the POV of the SDKs
+
+		// It is valid for a credential provider to not return an expiration;
+		// however, we need to have an expiration if a token is present to
+		// satsify various client SDKs. In this case, we return an expiration
+		// timestamp a fixed point in the future.
+		// https://github.com/awslabs/amazon-ecs-local-container-endpoints/issues/26
+		if err != nil && len(response.Token) > 0 {
```

Comment:
Curious- why check the length of Token?

Reply:
So we don't set the expiration for regular (IAM user, non-expiring) access/secret key. Token is a string field, so it's never nil; just empty (`len() == 0`) if it's not present.